pytorch_lightning/loops/utilities.py

```diff
@@ -40,76 +39,51 @@ def check_finite_loss(loss: Optional[torch.Tensor]) -> None:
 
 
 def _check_training_step_output(model: "pl.LightningModule", training_step_output: STEP_OUTPUT) -> None:
-    """Sanity checks that training produced a valid output and optimizer step has already been called in manual
-    optimization.
+    """Sanity checks that training produced a valid output.
 
     Args:
         model: a reference to the trainer
         training_step_output: the output of the training step (before wrapping in an AttributeDict)
     """
-    if isinstance(training_step_output, torch.Tensor) and not model.automatic_optimization:
-        if training_step_output.grad_fn is None:
-            # TODO: Find why - RuntimeError: Expected to mark a variable ready only once ...
-            raise MisconfigurationException("In manual optimization, `training_step` should not return a Tensor")
-    elif model.automatic_optimization:
-        if not any(
-            (
-                isinstance(training_step_output, torch.Tensor),
-                (isinstance(training_step_output, Mapping) and "loss" in training_step_output),
-                training_step_output is None,
-            )
-        ):
+    if (
+        isinstance(training_step_output, torch.Tensor)
+        and not model.automatic_optimization
+        and training_step_output.grad_fn is None
+    ):
+        # TODO: in manual optimization, anything returned should be considered an `extra`
+        raise MisconfigurationException("In manual optimization, `training_step` should not return a Tensor")
+    if model.automatic_optimization and not (
+        isinstance(training_step_output, torch.Tensor)
+        or (isinstance(training_step_output, Mapping) and "loss" in training_step_output)
+        or training_step_output is None
+    ):
+        raise MisconfigurationException(
+            "In automatic optimization, `training_step` must either return a Tensor, "
+            "a dict with key 'loss' or None (where the step will be skipped)."
+        )
+
+
+def _extract_hiddens(training_step_output: STEP_OUTPUT, truncated_bptt_steps: int) -> Optional[Any]:
+    """Get the hidden embeddings if present from the training step output.
```

Comment:
```suggestion
    """Get the hidden state if present from the training step output.
```
to assume these are embeddings is a bit much haha